fsspec/mapping.py

```diff
@@ -37,7 +37,10 @@ class FSMap(MutableMapping):
 
     def __init__(self, root, fs, check=False, create=False, missing_exceptions=None):
         self.fs = fs
-        self.root = fs._strip_protocol(root).rstrip("/")
+        try:
+            self.root = fs._strip_protocol(root, remove_trailing_slash=True)
+        except TypeError:
```

Comment:
This is unfortunate; in what case does `rstrip` fail? In this case, I think we might be explicitly joining on the separator anyway.

Reply:
On Windows if the root is something like `c:/`, `rstrip` will cause a root of `c:` which is treated like '.', the current working directory instead of the root.
On posix `rstrip` will make `/` change to ` `.

Reply:
This is the only place that the special case for LocalFileSystem leaks, and I don't like it :|. I think it may be better to let through strange behaviour if the user thinks they want to may their whole C drive (and gets only cwd files instead).

Reply:
I've changed this back.

Reply:
But without `.rstrip("/")`